doc/dbus_api.md

```diff
@@ -111,66 +111,327 @@ Iface: o.o.YaST.Installer1.Software
+
+#### `org.opensuse.DInstaller.Storage1.Proposal` Interface
+
+Information about the calculated storage proposal.
+
+##### Properties
+
+~~~
+CandidateDevices    readable as
+LVM                 readable b
+EncryptionPassword  readable s
+Volumes             readable aa{sv}
+Actions             readable aa{sv}
+~~~
+
+##### Details
+
+###### `Volumes` Property
+
+~~~
+Volumes             readable aa{sv}
+~~~
+
+List of volumes used for calculating the proposal.
+
+Each volume is defined by the following properties:
+
+~~~
+DeviceType                s
+Optional                  b
+Encrypted                 b
+MountPoint                s
+FixedSizeLimits           b
+AdaptativeSizes           b
+MinSize                   x
+MaxSize                   x
+FsTypes                   as
+FsType                    s
+Snapshots                 b
+SnapshotsConfigurable     b
+SnapshotsAffectSizes      b
+VolumesWithFallbackSizes  as
+~~~
+
+Example:
+
+~~~
+1 14 DeviceType s "partition" Optional b false Encrypted b false MountPoint s / FixedSizeLimit b false AdaptiveSizes b false MinSize x 1024 MaxSize x 2048 FsTypes as 3 Btrfs XFS EXT4 FsType Btrfs Snapshots b true SnapshotsConfigurable b true SnapshotsAffectSizes b false VolumeWithFallbackSizes as 1 /home
+~~~
+
+###### `Actions` Property
+
+~~~
+Actions             readable aa{sv}
+~~~
+
+Actions to perform in the system to create the proposal. If the proposal failed, then the list of actions is empty.
+
+Each action is defined by the following properties:
+
+~~~
+Text    readable s
+Subvol  readable b
+Delete  readable b
+~~~
+
+Example:
+
+~~~
+2 3 Text s "Create partition /dev/vdb1" Subvol b false Delete b false 3 Text s "Delete Btrfs subvolume @/var" Subvol b true Delete b true
+~~~
+
+#### `org.opensuse.DInstaller.Storage1.ISCSI.Initiator` Interface
+
+Provides methods for configuring iSCSI initiator and for discovering nodes.
+
+##### Methods
+
+~~~
+Discover(in  s       address,
+         in  u       port,
+         in  a{sv}   options,
+         out u       result)
+Delete(in o  iscsi_node_path,
+       out u result)
+~~~
+
+##### Properties
+
+~~~
+IniciatorName readable,writable   s
+~~~
+
+##### Details
+
+###### `Discover` Method
+
+~~~
+Discover(in  s       address,
+         in  u       port,
+         in  a{sv}   options,
+         out u       result)
+~~~
+
+Performs nodes discovery. Discovered nodes are exported with the path `/org/opensuse/DInstaller/iscsi/node[0-9]+`.
+
+Arguments:
+
+* `in s address`: IP address of the iSCSI server.
+* `in u port`: Port of the iSCSI server.
+* `in a{sv} options`:
+  * `InitiatorUsername s`: Username for incoming authentication.
+  * `InitiatorPassword s`: Password for incoming authentication.
+  * `TargetUsername s`: Username for outgoing authentication.
+  * `TargetPassword s`: Password for outgoing authentication.
```

Comment:
I fear the terms "incoming" and "outgoing" regarding iSCSI authentication. They are not so well established and I find they can easily lead to confusion.

I guess with "incoming authentication" you mean the Target authenticates the initiator and with "outgoing" you mean the initiator authenticates the target.

But then in the code below you do this:

```ruby
Y2IscsiClient::Authentication.new.tap do |auth|
  auth.username = target_username
  auth.password = target_password
  auth.username_in = initiator_username
  auth.password_in = initiator_password
end
```

Which is the opposite. `Y2IscsiClient::Authentication#username_in` is documented as "_Username for the CHAP challenge during authentication by the initiator_". That would be "outgoing authentication" in your jargon, so you should have assigned `TargetUsername` to `username_in`.